internal/controllers/synthesis/lifecycle.go

```diff
@@ -115,13 +106,17 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, nil
 		}
 
+		// TODO: Is it possible to delete before the first synthesis?
+
 		// Deletion increments the composition's generation, but the reconstitution cache is only invalidated
 		// when the synthesized generation (from the status) changes, which will never happen because synthesis
 		// is righly disabled for deleted compositions. We break out of this deadlock condition by updating
 		// the status without actually synthesizing.
 		if comp.Status.CurrentState != nil && comp.Status.CurrentState.ObservedCompositionGeneration != comp.Generation {
 			comp.Status.CurrentState.ObservedCompositionGeneration = comp.Generation
-			comp.Status.CurrentState.Synthesized = true // in case the previous synthesis failed
+			comp.Status.CurrentState.Ready = false
+			comp.Status.CurrentState.Reconciled = false
+			comp.Status.CurrentState.Synthesized = true // in case the previous synthesis failed (TODO I don't think this actually works)
```

Comment:
Interesting, what is this supposed to do?

Reply:
I don't remember lol 😆 . I'll follow up on the TODO though and leave more comments